vendor/src/github.com/docker/swarmkit/ca/server.go

```diff
@@ -233,8 +233,9 @@ func (s *Server) IssueNodeCertificate(ctx context.Context, request *api.IssueNod
 					},
 				},
 				Spec: api.NodeSpec{
-					Role:       role,
-					Membership: api.NodeMembershipAccepted,
+					Role:         role,
+					Membership:   api.NodeMembershipAccepted,
+					Availability: request.Availability,
```

Comment:
@aaronlehmann The `request.Availability` is defined as 
```
type NodeSpec_Availability int32
```
It seems the default value will be `0` so not sure we could check if this field is set or not.